File: miniJobs/Domain/Entities/Employer.cs
using Domain.Interfaces;

namespace Domain.Entities;
[Table("employers")]
public class Employer : IEntity<int>
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("name")]
    public string? Name { get; set; }

    [Column("id_number")]
    public string? IdNumber { get; set; }

    [Column("company_phone_number")]
    public string? CompanyPhoneNumber { get; set; }


    [Column("street_address_and_number")]
    public string StreetAddressAndNumber { get; set; }

    [NotMapped]
    public User User { get; set; }
}

File: miniJobs/UI/minijobs_admin/lib/models/employer.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:minijobs_admin/models/user.dart';
part 'employer.g.dart';

@JsonSerializable()
class Employer extends User {
  String? name;
  String? streetAddressAndNumber;
  String? idNumber;
  String? companyPhoneNumber;

  Employer(
      super.id,
      super.firstName,
      super.lastName,
      super.userName,
      super.email,
      super.phoneNumber,
      super.role,
      super.deleted,
      super.accountConfirmed,
      this.name,
      this.streetAddressAndNumber,
      this.idNumber,
      this.companyPhoneNumber);

       factory Employer.fromJson(Map<String, dynamic> json) => _$EmployerFromJson(json);

  Map<String, dynamic> toJson() => _$EmployerToJson(this);
}
